esp-hal/src/dma/pdma.rs

```diff
@@ -57,20 +47,9 @@ macro_rules! ImplSpiChannel {
                     #[cfg(esp32)]
                     {
                         let dport = unsafe { &*crate::peripherals::DPORT::PTR };
-
-                        match $num {
-                            2 => {
-                                dport
-                                .spi_dma_chan_sel()
-                                .modify(|_, w| unsafe { w.spi2_dma_chan_sel().bits(1) });
-                            },
-                            3 => {
-                                dport
-                                .spi_dma_chan_sel()
-                                .modify(|_, w| unsafe { w.spi3_dma_chan_sel().bits(2) });
-                            },
-                            _ => panic!("Only SPI2 and SPI3 supported"),
-                        }
+                        dport
```

Comment:
This `init_channel` function can be removed and this code can be moved to either `configure`/`configure_for_async` or `set_peripheral`.

Reply:
I didn't want to modify this code too much lest your PR will probably conflict, but let's see :)